src/types/questionnaire/validation.ts

```diff
@@ -5,7 +5,7 @@ import { QuestionValidationError } from "@/types/questionnaire/batch";
 export interface FieldMetadata<T = unknown> {
   key: string;
   required: boolean;
-  validate?: (value: T) => boolean;
+  validate?: (value: T) => boolean | string;
```

Comment:
since we are now doing try-catch method, this needs to be updated and other structured question's validates must be updated as well to follow this approach right?

```suggestion
  validate?: (value: T) => void;
```